scipy/stats/kde.py

```diff
@@ -619,6 +625,132 @@ def logpdf(self, x):
                                       self._norm_factor)
 
         return result
+      
+    def pdf_marginal(self, x, axis):
+        """
+        Evaluate the estimated marginal pdf on a provided set of points.
+        
+        Parameters
+        ----------
+        points : (# of dimensions, # of points)-array
+            Alternatively, a (# of dimensions,) vector can be passed in and
+            treated as a single point.
+        axis : int or 1-d array-like
+            Axis (axes) along which marginal pdf is evaluated. In the case of
+            1-d array-like, the elements should also be integers.
+        
+        Returns
+        -------
+        values : (# of points,)-array
+            The values at each point.
+        
+        Raises
+        ------
+        ValueError : if the dimensionality of the input points is larger than
+                     the dimensionality of the KDE, or if it's different from
+                     the length of axis, or if the input axis is not consisted 
+                     of integers.
+        
+        """
+        points = atleast_2d(x)
+        ax = atleast_1d(axis)
+
+        if not issubclass(ax.dtype.type, numbers.Integral):
+            msg = "elements of axis should be intergers"
+            raise ValueError(msg)
+
+        d, m = points.shape
+        if ax.size > self.d or ax.size == 0:
+            msg = "axis have dimension %s, dataset has dimension %s" % \
+                (ax.size, self.d)
+            raise ValueError(msg)
+
+        if d != ax.size:
+            if d == 1 and m == ax.size:
+                # points was passed in as a row vector
+                points = reshape(points, (m, 1))
+                d = ax.size
+                m = 1
+            else:
+                msg = "points have dimension %s, axis has dimension %s" % \
+                (d, ax.size)
+
+        cov = self.covariance[ax[:, newaxis], ax]
+        inv_cov = linalg.inv(cov)
+        
+        whitening = linalg.cholesky(inv_cov).T
```

Comment:
Get the cholesky factor of covariance `R` (hence decompose and invert instead of invert and decompose) and use `linalg.solve(R, np.hstack([self.dataset[ax], points])` for better numerical stability. Then you can slice it later. 

Generic comment: we have dropped Python 2 support hence you can use `@` for matmul.

Reply:
I adapted it from `evaluate`.  I'll do the same for that.

Reply:
> Get the cholesky factor of covariance `R` (hence decompose and invert instead of invert and decompose) and use `linalg.solve(R, np.hstack([self.dataset[ax], points])` for better numerical stability. Then you can slice it later.
> 
> Generic comment: we have dropped Python 2 support hence you can use `@` for matmul.

I think it's better to first slice it, then decompose and solve?

Reply:
Let me expand on my comment. Admittedly it was a bit cryptic. What I meant is as follows; let's take `cov = R @ R.T` then what you do here is get the `inv_cov = R^{-T} @ R{-1}` and R^{-1} is still lower triangular and still the cholesky factor of `inv_cov` hence `whitening = R^{-T}`

Then what you perform is
```[scaled_dataset, scaled_points] = R^{-T} @ [self.dataset[ax], points]```

Hence you can instead do 
```
zzz = linalg.solve_triangular(R.T, [(self.dataset[ax], points) hstacked], lower=True)
scaled_dataset, scaled_points = zzz[:. slice dimensions], zzz[:, the rest of it]
```

So you can get away without explicitly inverting any array. I hope I didn't make any mistakes though. Please check my argument.

Reply:
OK, I see your point.

Reply:
Ilayn, related general question
Is solve_triangular still faster than computing the inv cholesky , if this whitening has to be done repeatedly? 
Once we have the cholesky of the inverse we only need dot instead of solve_triangular.